java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java

```diff
@@ -17,7 +17,13 @@
 
 package org.apache.arrow.memory;
 
-
+/**
+ * Indicates memory could not be allocated for Arrow buffers.
+ *
+ * <p>This is different from {@linkplain OutOfMemoryError} which indicates the JVM
+ * is out of memory.  However, allocation of buffers is generally capped to a fixed
+ * size in application level code.
```

Comment:
I am not sure if the "however" part is necessary.

Reply:
We can probably mention that this errors indicates that we have run out of direct memory (which is a static limit) on the allocator when it is created.

Reply:
reworded.